app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java

```diff
@@ -0,0 +1,440 @@
+
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        startThreadUpdateRotation();
+    }
+
+    @Override
+    protected void onResume() {
+        mPlayer.setView(mViewPLayer);
+        super.onResume();
+    }
+
+    @Override
+    protected void onPause() {
+        mPlayer.removeView();
+        super.onPause();
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        stopThreadUpdateRotation();
+    }
+
+    @Override
+    protected void onDestroy() {
+        mStream.releaseStream();
+        if (mPlayer != null)
+            mPlayer.releasePlayer();
+        super.onDestroy();
+    }
+
+    private void startThreadUpdateRotation() {
+        stopThreadUpdateRotation();
+        if (mStream == null || mStream.getStructure() != IBandStream.Structure.EQUIRECTANGULAR)
+            return;
+        mIsThreadPositionActive = true;
+        mThreadUpdatePosition = new Thread() {
+            public Runnable mRunnableUpdateRotation = new Runnable() {
+                @Override
+                public void run() {
+                    mViewDirection.setRotation((float) Math.toDegrees(mViewPLayer.getPanoramaPositionAngle()));
+                }
+            };
+
+            @Override
+            public void run() {
+                while (mIsThreadPositionActive) {
+                    try {
+                        Thread.sleep(50);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    mViewDirection.post(mRunnableUpdateRotation);
+                }
+            }
+        };
+        mThreadUpdatePosition.start();
+
+    }
+
+    private void stopThreadUpdateRotation() {
+        mIsThreadPositionActive = false;
+        if (mThreadUpdatePosition != null)
+            mThreadUpdatePosition.interrupt();
+        mThreadUpdatePosition = null;
+    }
+
+    private void updateViewPlayPause() {
+        if (mPlayer.getPlayWhenReady())
+            mBtnPlayPause.setImageResource(R.drawable.ic_player_pause);
+        else
+            mBtnPlayPause.setImageResource(R.drawable.ic_player_play);
+    }
+
+    private void toast(String message) {
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
+
+
+    }
+
+    private void onVariantSelected(int position) {
+        if (mVariants.get(position).isAuto()) {
+            mVariantAdapter.setSelected(position);
+            mPlayer.setAutoVariant();
+        } else {
+            mVariantAdapter.setSelected(position);
+            mPlayer.setCurrentVariant(mVariants.get(position).getIndex());
+        }
+        mVariantAdapter.notifyDataSetChanged();
+    }
+
+    private void setDisplayMode() {
+        if (mViewPLayer.getState() == IBandPlayerView.DisplayState.EQUIRECTANGULAR && mStream.getState() == IBandStream.State.OPEN) {
+            if (mViewPLayer.getVRMode()) {
+                mBottomActionBar.setVisibility(View.GONE);
```

Comment:
Move those commands to an informative function such as `setupEquirectangularVRView()`

Reply:
done